tools/shared.py

```diff
@@ -127,6 +127,17 @@ def run_process(cmd, check=True, input=None, *args, **kw):
   sys.stderr.flush()
   kw.setdefault('text', True)
   kw.setdefault('encoding', 'utf-8')
+  # Handle calls to the Python scripts by going through the python executable
+  if cmd and cmd[0].endswith('.py'):
+    cmd = [sys.executable, '-E'] + cmd
+    # $PYTHON -E will not ignore _PYTHON_SYSCONFIGDATA_NAME, which is an internal
+    # of cpython used in cross compilation via setup.py.
+    environ = kw.get('env')
+    if environ is None:
+      environ = os.environ.copy()
+    if '_PYTHON_SYSCONFIGDATA_NAME' in environ:
+      del environ['_PYTHON_SYSCONFIGDATA_NAME']
```

Comment:
How about if we simple did `del os.environ['_PYTHON_SYSCONFIGDATA_NAME']` somewhere at the top level instead of duplicating the environment here?

Reply:
Because users never call the Python scripts directly, we're not supposed to have `_PYTHON_SYSCONFIGDATA_NAME` in the environment anyway (since the batch wrappers unset it), but I was trying to be careful and handle the case of a potential call that uses a custom environment and accidentally sets it. I didn't find such a case in the code, but I wanted to handle it just in case I missed something. If that's not a case that should be handled, I'm finding with doing the removal at the top level, but I would appreciate some guidance as to best place for this code.

Reply:
I think the `get_command_with_possible_response_file` approach when calling the file_packager might actually be the best/simplest solution for now.

Reply:
Indeed, there's still a limit even with a direct process call without going through the batch file, but it is much higher (32KiB vs. 8KiB for Windows and CMD, respectively) so it will also help in the general case, and `file_packager` can additionally gain support for response files in order to work with even longer command lines.